libr/core/file.c

```diff
@@ -312,7 +312,13 @@ static bool setbpint(RCore *r, const char *mode, const char *sym) {
 	if (bp) {
 		bp->internal = true;
 #if __linux__
-		bp->data = r_str_newf ("?e %s: %s;dd", mode, sym);
+		if (strncmp (sym, "sym.imp.dlclose", strlen ("sym.imp.dlclose"))) {
+			bp->data = r_str_newf ("?e %s: %s;ps@r:%s", mode, sym, 
+				!strncmp (sym, "sym.imp.dlmopen", strlen ("sym.imp.dlmopen"))
+				? "rsi" : "rdi");
```

Comment:
this is not portable. r2 have a way to determine the value of the register depending on the purpose, rsi/rdi is only for x86-64, and core/file.c is not the place to put arch-specific things